c7n/resources/cloudtrail.py

```diff
@@ -124,6 +124,50 @@ def __call__(self, r):
         return self.match(r['c7n:TrailStatus'])
 
 
+@CloudTrail.filter_registry.register('event-selectors')
+class EventSelectors(ValueFilter):
+    """Filter a cloudtrail by its related Event Selectors.
+    :Example:
+    .. code-block:: yaml
+        policies:
+          - name: cloudtrail-event-selectors
+            resource: aws.cloudtrail
+            filters:
+            - type: event-selectors
+              key: EventSelectors[].IncludeManagementEvents
+              op: contains
+              value: True
+    """
+
+    schema = type_schema('event-selectors', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('cloudtrail:GetEventSelectors',)
+    annotation_key = 'c7n:TrailEventSelectors'
+
+    def process(self, resources, event=None):
+        for r in resources:
+            region = self.manager.config.region
+            trail_arn = Arn.parse(r['TrailARN'])
+
+            if r.get('HomeRegion') and r['HomeRegion'] != region:
```

Comment:
we should extract a trail region client as a function in this module, return a region client and a set of trail resources within that region, it would address some code coverage gaps in this pr and extract some copy / paste in this module.

Reply:
I'm not sure I understand, do you mean something similar to this?
```
// Module level singleton
region_clients = {} 

def get_region_client(region):
    if region not in region_clients:
        region_clients[region] = session.client(
            'cloudwatch', region_name=region)
    
    return region_clients[region]

// Example usage in filter
def process (self, resources, event=None):
    for r in resources:
        trail_arn = Arn.parse(r['TrailARN'])
        client = get_region_client(trail_arn.region)
        // use client here
```

Reply:
not so much module level, more like a helper function that this filter and the other filters can use, alternatively a mixin subclass though a function is likely easier,  we can cross account boundaries between policy executions so we don't want to store at a global scope, we can group the trails by region when processing to avoid extraneous region client creation.